### ISSUE TITLE
Fix ostream test for macos

### ISSUE THREAD

**Issue by eseiler** [seqan/seqan3#723]
In the `gz_ostream_test.cpp` we write:

https://github.com/seqan/seqan3/blob/1264624bf6da2b7743587a23ee9eb7797ee13f02/test/unit/contrib/stream/gz_ostream_test.cpp#L20-L26

The first 10 bytes are flags:
```
'\x1f','\x8b','\x08','\x00','\x00','\x00','\x00','\x00','\x00','\x03'
  ID1    ID2    CM     FLG    MTIME  MTIME  MTIME  MTIME  XFL    OS
```

with the following meaning:

| Flag  | Meaning                          |
|-------|----------------------------------|
| ID1   | Magic byte, fixed to 1f          |
| ID2   | Magic byte, fixed to 8b          |
| CM    | Compression method, here DEFLATE |
| MTIME | 4 byte time stamp                |
| XFL   | extra flags                      |
| OS    | Operating System                 |

The operating system flag is set to 03 for unix.
When running it on MacOS it may be set to 03, 07 (for macos <10) or 13 (macos > 10).

So I think we shouldn't check the 'correctness' of the OS flag?

**Comment by h-2** [comment]
This is also related to #857

**Comment by h-2** [comment]
Can you make a fix, @eseiler ? Either make OS specific checks or a wildcard somehow at the position.

**Comment by eseiler** [comment]
Fixed by #1014
